enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,597 @@
+but diverges with [Ingress Mutable Publishing Scope](/enhancements/ingress/mutable-publishing-scope.md).
+The mutable publishing scope design set `status.endpointPublishingStrategy.loadBalancer.scope`
+to the initial value of `spec.endpointPublishingStrategy.loadBalancer.scope` so that it could
+detect when the spec for `Scope` changed. Therefore, `status.endpointPublishingStrategy.loadBalancer.scope`
+may not always reflect the actual scope of the load balancer. This proposal for `Subnets`
+perpetuates the pattern of ensuring that `status.endpointPublishingStrategy` reflects the
+_actual_ value (in our case, the annotation value).
+
+#### Install Time Configuration
+
+TODO
+
+### Risks and Mitigations
+
+#### Invalid Subnet Annotation Values
+
+One type of risk is if the `service.beta.kubernetes.io/aws-load-balancer-subnets`
+annotation is configured with invalid subnets. In this situation, the AWS Cloud
+Controller Manager (CCM) will throw an error and not reconcile the service. 
+
+This means that if a service is created (or recreated) with an invalid annotation,
+the load balancer will not be provisioned. If an existing service has an invalid
+annotation added, any future updates to the service will remain unreconciled until
+the invalid subnet is fixed.
+
+The following are examples of invalid values that can cause an error in the AWS CCM:
+
+1. Multiple Subnets in Same Availability Zone (AZ)
+2. Non-Existent Subnets
+3. Replacing All Subnets (disjoint union) in a Classic Load Balancer (CLB)
+
+Since #1 and #2 can occur during service creation, they remain valid risks.
+See [Support Procedures](#support-procedures) for how to debug #1 and #2.
+
+However, we have completely mitigated #3 by requiring services to be recreated
+upon updating `Subnets` (see [Effectuating Subnet Updates](#effectuating-subnet-updates)).
+See the next section, [Replacing All Subnets in a CLB](#replacing-all-subnets-in-a-clb)
+for more details on #3.
+
+##### Replacing All Subnets in a CLB
+
+If an update to the list of subnets in `service.beta.kubernetes.io/aws-load-balancer-subnets`
+would completely replace the existing subnets on an existing service
+using a CLB, the AWS CCM will throw an error and not reconcile the service.
+As outlined in this [GitHub comment](https://github.com/kubernetes/cloud-provider-aws/issues/249#issuecomment-906595623),
+a new set of subnets causes the AWS controller attempts to detach all
+existing subnets on CLBs which is not a supported operation.
+
+A cluster admin wanting to replace all subnets would need to delete and
+recreate the LoadBalancer-type service. It may not be obvious that to the cluster
+admin that they are replacing all subnets because they must be aware of which
+subnets are currently used by the load balancer. This includes automatically
+selected subnets when NOT using `service.beta.kubernetes.io/aws-load-balancer-subnets`.
```

Comment:
If this is true?
```suggestion
subnets are currently used by the load balancer. For example, if the subnets were automatically
selected subnets they are not listed anywhere.
```

Reply:
Yea that's a fair, the cluster admin doesn't know which subnets they are using. Done.